src/Writer/BatchWriter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Ddeboer\DataImport\Writer;
+
+use Ddeboer\DataImport\Writer;
+
+/**
+ * @author Markus Bachmann <markus.bachmann@bachi.biz>
+ */
+class BatchWriter implements Writer
+{
+    private $delegate;
+
+    private $size;
+
+    private $queue;
+
+    public function __construct(Writer $delegate, $size = 20)
+    {
+        $this->delegate = $delegate;
+        $this->size = $size;
+    }
+
+    public function prepare()
+    {
+        $this->delegate->prepare();
+
+        $this->queue = new \SplQueue();
```

Comment:
Are you using SplQueue because of memory efficiency? Is it confirmed it uses less memory?